android/KMEA/app/src/main/res/values/strings.xml

```diff
@@ -58,6 +58,8 @@
     <string name="checking_model_download" translatable="false">Checking to download dictionary&#8230;</string>
     <string name="downloading_model" translatable="false">Downloading dictionary&#8230;</string>
     <string name="catalog_unavailable" translatable="false">The resource catalog is unavailable\n</string>
+    <string name="catalog_download_start_in_background" translatable="false">Catalog update started in background.\n</string>
+    <string name="catalog_download_is_running_in_background" translatable="false">The download of catalog is running. Please, try again later!\n</string>
```

Comment:
```suggestion
    <string name="catalog_download_is_running_in_background" translatable="false">The catalog is still downloading; please try again in a moment!\n</string>
```

Ideally, if the catalog is downloading when the user tries to get access to it, Keyman should show a UI widget that waits for the download to finish before dismissing automatically and taking the user to the next point in the process. This way, the download starts automatically in the background before the user requests it, and hopefully it will be done by the time they need it, but they don't have to repeatedly try to see if it is done ...

Reply:
Christopher suggested to trigger the catalog update during startup of the application.
I didn't want to separate the Downloading the catalog from fetch catalog in the application.

Maybe I should do that in a second step? 
- Separate Download from Fetch catalog
- Adding an Alert dialog which informs the user, when Download is in progress?
- Start download automatically on startup of the application
- Inform user when opening language settings, that download is necessary and allow to restart the update manually in language settings page

Reply:
That sounds good to me; I'm not sure about:

> Inform user when opening language settings, that download is necessary and allow to restart the update manually in language settings page

If the app is downloading the catalog automatically on startup, this shouldn't be required?

But, I am unsure about starting the catalog download automatically when opening the application; many of our users are on expensive connections so it is not ideal to do this for the whole huge catalog. However, there is a question about checking for updates; we'd only check for updates for the keyboards that they have installed, ideally.

Reply:
OK maybe ask the user before starting the update on startup?
Something like "Your language and keyboard data" might be out of date. Do you want to start an update?"
Because if the internet connection is very slow the user have to wait a for long time when you start update only when the user tries to access the catalog.

Reply:
I don't think we can do anything that pops a dialog up in front of the user, because that hurts usability.

For now, let's go with starting the download when the app starts (do this update no more than once a week). Then make sure we have an option in the menu somewhere to re-download, which should show an "updating..." banner but doesn't block the UI.

Reply:
Is the notification enough to inform the user, when the update is in progress?

Reply:
> (do this update no more than once a week)

Originally, it was a 1-hour threshold. Any thoughts on making this 1 day? I'm thinking of people submit a new keyboard in the repo who may not want to wait a week to download their keyboard.

Reply:
I think 1 hour is the time for reload the file cache.
I think one week for automatically update might be OK, when we add a menuicon for manual update

Reply:
I found out I can not move the initialize data step to application startup until we changed the keyboard download because the update of keyboards and lexical models will update the keyboard resources as well.

Reply:
We need to move this step after changing key board downloader.
The behavior should now be consistent right now.

Check for about in LanguagesSettings like before.
On all other access points to the Catalog you get the "Download in Progress"-Message as long as it is finished.

Reply:
On iOS the preferred way to force refresh is to drag the menu down until a spinner appears. Is this the same for Android?

Reply:
I think only when the reload is finished after the spinner is closed. In our case you will get the notification: "Download started" or "Download is running"

Reply:
Postponed to a new Pull Request